src/backend/storage.py

```diff
@@ -0,0 +1,66 @@
+import os
+import posixpath
+import uuid
+from typing import Literal
+
+from dotenv import load_dotenv
+from fastapi import HTTPException, UploadFile
+from storage3 import AsyncStorageClient, create_client
+from storage3.utils import StorageException
+
+ALLOWED_MIMETYPES = {"image/jpeg", "image/png"}
+MAX_SIZE = 50 * 2**20  # 50 MB
+BUCKET_NAME = "user-post"
+
+load_dotenv()
+SUPABASE_URL = os.getenv("SUPABASE_URL")
+SUPABASE_API_KEY = os.getenv("SUPABASE_API_KEY")
+headers = {"apiKey": SUPABASE_API_KEY, "Authorization": f"Bearer {SUPABASE_API_KEY}"}
+
+storage_client: AsyncStorageClient = create_client(
+    SUPABASE_URL, headers=headers, is_async=True
+)  # type:ignore
```

Comment:
why'd we need a type ignore here?

Reply:
The type ignore is there because `SUPABASE_URL`  will be `None` if the env var isn't set- and those parameters in `create_client` have to be `str`. I added the type ignore temporarily because I was supposed to check if the environment variables have been set. But I've forgotten to, oopsie.

Reply:
👍🏼 there was another type-ignore (i think in `.upload()`) that can be removed soon because I'd fixed the typehint in the lib, idk when I'll make a new release though